xgi/classes/dihypergraph.py

```diff
@@ -0,0 +1,610 @@
+"""Base class for undirected hypergraphs."""
+from collections import defaultdict
+from collections.abc import Hashable, Iterable
+from copy import copy, deepcopy
+from itertools import count
+from warnings import warn
+
+from ..exception import XGIError
+from ..utils import IDDict, update_uid_counter
+from .reportviews import EdgeView, NodeView
+
+__all__ = ["DiHypergraph"]
+
+
+class DiHypergraph:
+    r"""A hypergraph is a collection of subsets of a set of *nodes* or *vertices*.
```

Comment:
Not sure you want comments on the docstring at this time, but this seems to be copy/pasted from the Hypergraph class, no?

Reply:
Yes, that's correct. I still have yet to update the docs and add unit tests.